src/IECoreScene/MeshPrimitive.cpp

```diff
@@ -454,6 +527,23 @@ void MeshPrimitive::load( IECore::Object::LoadContextPtr context )
 	{
 		removeCreases();
 	}
+
+	std::string interpolateBoundary, faceVaryingLinearInterpolation, triangleSubdivisionRule;
+	if( container->hasEntry( g_interpolateBoundaryEntry ) )
+	{
+		container->read( g_interpolateBoundaryEntry, interpolateBoundary );
+		setInterpolateBoundary( interpolateBoundary );
+	}
+	if( container->hasEntry( g_faceVaryingLinearInterpolationEntry ) )
+	{
+		container->read( g_faceVaryingLinearInterpolationEntry, faceVaryingLinearInterpolation );
+		setFaceVaryingLinearInterpolation( faceVaryingLinearInterpolation );
+	}
+	if( container->hasEntry( g_triangleSubdivisionRuleEntry ) )
+	{
+		container->read( g_triangleSubdivisionRuleEntry, triangleSubdivisionRule );
+		setTriangleSubdivisionRule( triangleSubdivisionRule );
+	}
 }
 
 bool MeshPrimitive::isEqualTo( const Object *other ) const
```

Comment:
As well as the methods modified here, we also need to update `MeshPrimitive::hash()`.